src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,101 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import React, { useEffect, useState } from 'react';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import classNames from 'classnames';
+import { useParams } from 'react-router-dom';
+import { SortParam } from './SortParam';
+import { SortTypes } from '../types/SortTypes';
+
+type Props = {
+  people: Person[] | null;
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const sortKeys = [
+    SortTypes.Name,
+    SortTypes.Sex,
+    SortTypes.Born,
+    SortTypes.Died,
+  ];
+
+  const { personId } = useParams();
+  const [activePersonName, setActivePersonName] = useState<string | null>(null);
+
+  const isPersonInTable = (personName: string) => {
+    return people?.find(({ name }) => name === personName);
+  };
+
+  const handleLinkForParent = (personName: string) => {
+    if (!people) {
+      return;
+    }
+
+    if (isPersonInTable(personName)) {
+      const indexOfPerson = people?.findIndex(
+        person => person.name === personName,
+      );
+      const person = people[indexOfPerson];
+      const { name, sex, slug } = person;
+
+      return <PersonLink name={name} sex={sex} slug={slug} />;
+    }
+
+    return personName;
+  };
```

Comment:
The handleLinkForParent function can return either a React component (PersonLink) or a string. This inconsistency might cause rendering issues. Consider ensuring that the return type is consistent, possibly by always returning a React element.